File: src/chipschallenge/clonebehaviors/CloneBoss.java
package chipschallenge.clonebehaviors;

import chipschallenge.Block;
import chipschallenge.Creatures;

public class CloneBoss extends CloneBehavior {

    private CloneBoss() {
    }
    private static CloneBoss mInstance = null;

    public static synchronized CloneBoss getInstance() {
        if (mInstance == null) {
            mInstance = new CloneBoss();
        }
        return mInstance;
    }

    public Block cloneIt(Block original) throws CloneNotSupportedException {
        System.out.println("Clone Irene Boss");
        Block boss = Creatures.getBoss();
        if (boss == null) {
            throw new CloneNotSupportedException("No controller");
        } else {
            System.out.println(boss + " " + boss.getFacing());
            return cloneTo(original, boss.getFacing());
        }
    }
}

File: ChipsChallenge.Shared/Clonebehaviors/CloneBoss.cs
namespace ChipsChallenge.Shared.Clonebehaviors
{
    public class CloneBoss : CloneBehavior
    {
        private static CloneBoss instance;

        private CloneBoss()
        {
        }

        public static CloneBoss Instance
        {
            get
            {
                lock (typeof(CloneBoss))
                {
                    return instance ?? (instance = new CloneBoss());
                }
            }
        }

        public override Block CloneIt(Block original)
        {
            Block boss = Creatures.Boss;
            if (boss == null)
            {
                throw new CloneNotSupportedException("No boss");
            }

            return CloneTo(original, boss.Facing);
        }
    }
}